aztec-up/bin/aztec

```diff
@@ -207,6 +401,151 @@ case ${1:-} in
+
+EOF
+      exit 0
+    fi
+
+    # Check if PATH was provided
+    if [ -z "$PROJECT_PATH" ]; then
+      echo "Error: PATH argument is required"
+      echo "Usage: aztec new [OPTIONS] <PATH>"
+      echo "Run 'aztec new --help' for more information"
+      exit 1
+    fi
+
+    validate_working_directory
+
+    echo "Creating new Noir project at $PROJECT_PATH..."
+
+    # Run nargo new via docker
+    run_nargo "${NARGO_ARGS[@]}"
+
+    # Check if nargo new succeeded
+    if [ $? -ne 0 ]; then
+      >&2 echo "Project creation failed!"
+      exit 1
+    fi
+
+    # Setup Aztec contract project (add dependency and create template)
+    if [ "$IS_CONTRACT" = true ]; then
+      setup_aztec_contract_project "$PROJECT_PATH"
+    fi
+    ;;
+  init)
+    shift
+
+    # Parse arguments
+    parse_project_flags_and_build_nargo_args "init" "$@"
+    PARSE_RESULT=$?
+
+    # Handle help request
+    if [ $PARSE_RESULT -eq 2 ]; then
+      cat << 'EOF'
+Aztec Init - Create a new Aztec Noir project in the current directory
+
+Usage: aztec init [OPTIONS]
+
+Options:
+  --name <NAME>  Name of the package [default: current directory name]
+  --lib          Use a library template
+  --bin          Use a binary template
+  --contract     Use a contract template [default]
+  -h, --help     Print help
+
+This command creates a new Aztec Noir project in the current directory using nargo
+and automatically adds the Aztec.nr dependency to your Nargo.toml file.
+
+ENVIRONMENT VARIABLES:
+  AZTEC_PATH    Path to Aztec installation (default: $HOME/.aztec)
+  VERSION       Aztec version to use (default: from $AZTEC_PATH/default_version)
+  DOCKER_REPO   Docker repository (default: aztecprotocol/aztec)
+
+EOF
+      exit 0
+    fi
+
+    validate_working_directory
+
+    echo "Initializing Noir project..."
+
+    # Run nargo init via docker
+    run_nargo "${NARGO_ARGS[@]}"
+
+    # Check if nargo init succeeded
+    if [ $? -ne 0 ]; then
+      >&2 echo "Initialization failed!"
+      exit 1
+    fi
+
+    # Setup Aztec contract project (add dependency and create template)
+    if [ "$IS_CONTRACT" = true ]; then
+      setup_aztec_contract_project "."
+    fi
+    ;;
+  fmt)
+    shift
+
+    validate_working_directory
+
+    # Run nargo fmt
+    run_nargo fmt "$@"
+    ;;
+  check)
+    shift
+
+    validate_working_directory
+
+    # Run nargo check
+    run_nargo check "$@"
+    ;;
+  lsp)
+    # TODO: Drop this (and the corresponding docs in cli.ts) in case `nargo` is to be used natively (not containerized).
+
+    # Special-case nargo's LSP command:
+    # 1. include --rm for cleanup
+    # 2. don't specify a user (run as root)
+    # 3. don't specify a workdir
+    validate_working_directory
+
+    docker run --rm -i \
+      --name aztec-nargo-lsp \
+      -v $HOME:$HOME \
+      -e HOME=$HOME \
+      --entrypoint=/usr/src/noir/noir-repo/target/release/nargo \
+      $DOCKER_REPO:$VERSION lsp
+    ;;
```

Comment:
This is a copy-paste from the original `aztec-nargo` command. Might be dropped in the future if it's decided we don't want to use containerized `nargo`.